### PR TITLE
delete route by additional parameters

### PR DESCRIPTION
**Pull request by jxiaobin**
add new API to delete route by prefix, nexthop and policy.
nexthop and policy are optional.

### CODE REVIEW

File: logical_router_static_route.go [hunk 1]
```diff
@@ -36,14 +37,19 @@ func (odbi *ovndb) lrsrAddImp(lr string, ip_prefix string, nexthop string, outpu
...
+	if len(output_port) == 1 {
 		row["output_port"] = output_port
+	} else if len(output_port) > 1 {
+		return nil, fmt.Errorf("only 1 output port is allowed")
 	}
-	if len(policy) > 0 {
-		row["policies"] = policy
+	if len(policy) == 1 {
+		row["policy"] = policy
+	} else {
```
Comment (hzhou8) [inline logical_router_static_route.go:51]: else if len() > 1

File: logical_router_static_route.go [hunk 2]
```diff
@@ -36,14 +37,19 @@ func (odbi *ovndb) lrsrAddImp(lr string, ip_prefix string, nexthop string, outpu
...
-	if len(output_port) > 0 {
+	if len(output_port) == 1 {
 		row["output_port"] = output_port
+	} else if len(output_port) > 1 {
+		return nil, fmt.Errorf("only 1 output port is allowed")
 	}
-	if len(policy) > 0 {
-		row["policies"] = policy
+	if len(policy) == 1 {
+		row["policy"] = policy
```
Comment (hzhou8) [inline logical_router_static_route.go:50]: thanks for the fix. This part is better to be in a separate PR.

File: logical_router_static_route.go [hunk 3]
```diff
@@ -115,6 +121,50 @@ func (odbi *ovndb) lrsrDelImp(lr string, ip_prefix string) (*OvnCommand, error)
 	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
 }
 
+func (odbi *ovndb) lrsrDelImpEx(lr string, prefix string, nexthop, policy *string) (*OvnCommand, error) {
```
Comment (hzhou8) [inline logical_router_static_route.go:124]: better to rename as lrsrDelExImp

File: logical_router_static_route.go (hunk 1 above)
Comment (hzhou8) [reply logical_router_static_route.go:51]: Maybe it is better to use the same convention as LRSRDelEx(), i.e. make the parameter *string instead of []string

**Comment by hzhou8** [comment]
Please check CI failure as well

File: client.go [hunk 4]
```diff
@@ -99,6 +99,8 @@ type Client interface {
 	LRSRAdd(lr string, ip_prefix string, nexthop string, output_port []string, policy []string, external_ids map[string]string) (*OvnCommand, error)
 	// Delete LRSR with given ip_prefix on given lr
 	LRSRDel(lr string, ip_prefix string) (*OvnCommand, error)
+	// Delete LRSR with given ip_prefix, nexthop and policy on given lr
+	LRSRDelEx(lr string, prefix string, nexthop, policy *string) (*OvnCommand, error)
```
Comment (hzhou8) [inline client.go:103]: Sorry, one more comment here. We should add all parameters including output_port here, as optional parameter, i.e. compare if it is not nil. There can be routes with same prefix, nexthop, policy, but different output_ports. This API should allow user to delete just one of them.

Comment (noah8713) [reply client.go:103]: Why a need for new API instead of extending the existing LRSRDel? Default can always be delete by prefix if nexthop, policy, etc are not pass.

Comment (jxiaobin) [reply client.go:103]: @noah8713 we want to selectively delete route with the same prefix but different nexthop, policy.

File: logical_router_static_route.go (hunk 2 above)
Comment (jxiaobin) [reply logical_router_static_route.go:50]: ok

File: logical_router_static_route.go (hunk 3 above)
Comment (jxiaobin) [reply logical_router_static_route.go:124]: ok

**Comment by jxiaobin** [comment]
@hzhou8  revised as requested, could you plz take a look?

File: logical_router_static_route.go [hunk 5]
```diff
@@ -115,6 +117,53 @@ func (odbi *ovndb) lrsrDelImp(lr string, ip_prefix string) (*OvnCommand, error)
 	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
 }
 
+func (odbi *ovndb) lrsrDelExImp(lr string, prefix string, nexthop, policy, outputPort *string) (*OvnCommand, error) {
```
Comment (hzhou8) [inline logical_router_static_route.go:120]: Sorry, one more comment. It is better that the lrsrDelImp() just call this function now.

File: client.go (hunk 4 above)
Comment (hzhou8) [reply client.go]: @noah8713 Basically, to avoid breaking existing client for compatibility. If there is no other user using this API, it is better to just change the existing API. @jxiaobin maybe @noah8713 's suggestion makes sense because we just discussed in this PR regarding fixing the API LRSRAdd, which would have to break the existing API users.

File: logical_router_static_route.go (hunk 5 above)
Comment (jxiaobin) [reply logical_router_static_route.go:120]: fixed
